src/plugins/inputs/ignore.js

```diff
@@ -0,0 +1,89 @@
+"use strict";
+
+const Chan = require("../../models/chan");
+const Msg = require("../../models/msg");
+const Helper = require("../../helper");
+
+exports.commands = [
+	"ignore",
+	"unignore",
+	"ignorelist",
+];
+
+exports.input = function(network, chan, cmd, args) {
+	const client = this;
+	let target;
+	let hostmask;
+
+	if (cmd !== "ignorelist" && (args.length === 0 || args[0].trim().length === 0)) {
+		chan.pushMessage(client, new Msg({
+			type: Msg.Type.ERROR,
+			text: `Usage: /${cmd} <nick>[!ident][@host]`,
+		}));
+
+		return;
+	}
+
+	if (cmd !== "ignorelist") {
+		// Trim to remove any spaces from the hostmask
+		target = args[0].trim();
+		hostmask = Helper.parseHostmask(target);
+	}
+
+	switch (cmd) {
+	case "ignore":
+		if (hostmask.nick !== network.nick && !network.ignoreList.some(function(entry) {
```

Comment:
Send an error message when you try to ignore yourself instead of silently doing nothing.

It should also send an error when you try to ignore someone that is already ignored.